lb_content_resolver/content_resolver.py

```diff
@@ -277,3 +272,96 @@ def database_cleanup(self):
+                else:
+                    resolved_recordings.append({
+                        "artist_name": data["artist_name"],
+                        "recording_name": data["recording_name"],
+                        "recording_id": hit["recording_id"],
+                        "confidence": hit["confidence"],
+                        "index": data["index"],
+                    })
+
+            if len(next_query_data) == 0:
+                break
+
+            query_data = []
+            for data in next_query_data:
+                recording_name = mc.clean_recording(data["recording_name"])
+                if recording_name != data["recording_name"]:
+                    print(f'RETRY recording {data["recording_name"]} => {recording_name}')
+                    query_data.append({"artist_name": artist_name, "recording_name": recording_name, "index": data["index"]})
+
+                artist_name = mc.clean_artist(data["artist_name"])
+                if artist_name != data["artist_name"]:
+                    print(f'RETRY artist {data["artist_name"]} => {artist_name}')
+                    query_data.append({"artist_name": artist_name, "recording_name": recording_name, "index": data["index"]})
+
+            # If nothing got cleaned, we can finish now
+            if len(query_data) == 0:
+                break
+
+        return resolved_recordings
+
+    def resolve_playlist(self, jspf_playlist, m3u_playlist, match_threshold):
+        """ 
+            Given a JSPF playlist, resolve tracks and write the m3u file. Print output to console for now.
+        """
+        self.open_db()
+        self.build_index()
+
+        jspf = read_jspf_playlist(jspf_playlist)
+
+        title = jspf["playlist"]["title"]
+        recordings = []
+        artist_recording_data = []
+        for i, track in enumerate(jspf["playlist"]["track"]):
+            artist_recording_data.append({"artist_name": track["creator"], "recording_name": track["title"]})
+
+        hits = self.resolve_recordings(artist_recording_data, match_threshold)
+        hit_index = {hit["index"]: hit for hit in hits}
+
+        recording_ids = [r["recording_id"] for r in hits]
+        recordings = Recording.select().where(Recording.id.in_(recording_ids))
+        rec_index = {r.id: r for r in recordings}
+
+        results = []
+        for i, artist_recording in enumerate(artist_recording_data):
+            if i not in hit_index:
+                print("FAIL %s - %s not resolved." % (artist_recording["artist_name"], artist_recording["recording_name"]))
+                continue
+
+            hit = hit_index[i]
+            rec = rec_index[hit["recording_id"]]
+            hit["file_path"] = rec.file_path
+            hit["artist_name"] = rec.artist_name
+            hit["recording_name"] = rec.recording_name
+            results.append(hit)
+            print("OK   %s - %s resolved: %s" % (rec.artist_name, rec.recording_name, os.path.basename(rec.file_path)))
```

Comment:
Do we care that the output could be more readable? Currently it would be something like 

```
OK   Kendrick Lamar - Father Time resolved: Kendrick Lamar - 05 - Father Time.flac
```
where it's hard to see the final file name but it's fine if this is just for temporary debugging purposes. Similar point for the `FAIL` case above.

Reply:
It would be great to make it more readable -- I always struggle with this task, so if you have ideas, please!